Doppler.HtmlEditorApi/Storage.DapperProvider/LoadCampaignQuery.cs

```diff
@@ -0,0 +1,46 @@
+using System.Data;
+using System.Threading.Tasks;
+using Dapper;
+
+namespace Doppler.HtmlEditorApi.Storage.DapperProvider;
+
+public static class LoadCampaignQuery
+{
+    private const string QUERY = @"
+SELECT
+    CAST (CASE WHEN co.IdCampaign IS NULL THEN 0 ELSE 1 END AS BIT) AS CampaignHasContent,
+    CAST (CASE WHEN ca.IdUser IS NULL THEN 0 ELSE 1 END AS BIT) AS CampaignBelongsUser,
+    CAST (CASE WHEN ca.IdCampaign IS NULL THEN 0 ELSE 1 END AS BIT) AS CampaignExists,
+    ca.IdCampaign,
+    co.Content,
+    co.EditorType,
+    co.Meta
+FROM [User] u
+LEFT JOIN [Campaign] ca ON
+    u.IdUser = ca.IdUser
+    AND ca.IdCampaign = @IdCampaign
+LEFT JOIN [Content] co ON
+    ca.IdCampaign = co.IdCampaign
+WHERE
+    u.Email = @AccountName";
+
+    public static Task<Result> QueryFirstOrDefaultAsync(this IDbContext dbContext, Parameters param)
```

Comment:
Ya que sabemos qué hace el método, qué opinas? O perdemos algo?
```suggestion
    public static Task<Result> LoadCampaignAsync(this IDbContext dbContext, Parameters param)
```

Reply:
Mmm, si, es cierto, aunque se pierde algo de información. Que opinás de

```suggestion
    public static Task<Result> QueryFirstOrDefaultCampaignAsync(this IDbContext dbContext, Parameters param)
```

Osea, si Dapper usa el verbo `Query` en lugar de `Load`, me parece que es bueno mantenerlo.

`FirstOrDefault` indica que obtiene solo uno, y si no hay ninguno devuelve `null`.

Y luego nos queda `Campaign`, aunque en realidad, esto no es un `Campaign`, es un `LoadCampaignQuery.Result` (y no estoy seguro de que `LoadCampaignQuery` sea el nombre correcto para la clase estática)

Que difícil es poner nombres! Dudo!

![image](https://user-images.githubusercontent.com/1157864/154753928-8303144d-22ac-408d-8d0b-920b369cfdea.png)

Reply:
Creo que el nombre de la clase ya nos da el contexto que es campaña, nombrar el query de manera parecida no seria redundante?

Reply:
> Creo que el nombre de la clase ya nos da el contexto que es campaña, nombrar el query de manera parecida no seria redundante?

Así quedaría 
![image](https://user-images.githubusercontent.com/75738115/154754517-2f6ad8b2-5209-4c82-81b6-49c6dad9f5c3.png)

Puede que quede redundante ..

Reply:
Bueno, teniendo en cuenta esto y otras ideas de @ms-darianbenito cambié el approach, creo que ahora los nombres son mejores y sin el parche de la clase estática también estamos mejor.

El patron de query también es conocido, así que creo que es más intuitivo.

Resulevo la discución.